### PR TITLE
Env release clarification

### PR DESCRIPTION
**Pull request by beastawakens**

### CODE REVIEW

**Comment by beastawakens** [comment]
[Clarify env setting and release promotion in the docs](https://trello.com/c/nmnw5SEb/474-clarify-env-setting-and-release-promotion-in-the-docs)

**Review by camerondgray — APPROVED** [review]

File: docs/application/environment.md [hunk 1]
```diff
@@ -75,3 +75,9 @@ Release: RRJQTMVKRS
 ```
 
 `convox env edit` allows you to interactively update your environment variables in a terminal editor 😊
+
+<div class="block-callout block-show-callout type-info" markdown="1">
+If you update your env vars through the CLI, that will create a new release containing your changes.  For this to take effect, you will need to promote that release: `$ convox releases promote RRJQTMVKRS`
```
Comment (ddollar) [inline docs/application/environment.md]: Using a specific release ID can be confusing. What about just showing them `convox env set --promote`

Comment (beastawakens) [reply docs/application/environment.md]: I'll add that in, as that's super useful, but still think it's good to make a 2-step process clear as well.

**Review by camerondgray — CHANGES_REQUESTED** [review]
Overall I like where you are going but I think this structure is a bit confusing and it's a bit long. Since you are really overhauling things at this point I would suggest breaking things down something like:

- Explain that you can add env variables / secrets from the CLI or Web and the release process that goes with that
- Explain that if you have variables that are required for your app / service to function that you can declare them in the convox.yml at the app or service level
- Explain that you can give a declared variable a default value including a default value of blank

My personal opinion is to avoid variable names that match the use case (standard, required, etc...) because sometimes it cause even more confusion

**Review by camerondgray — APPROVED** [review]

**Pull request merged**